src/rust/engine/process_execution/src/lib.rs

```diff
@@ -248,7 +248,13 @@ pub struct Process {
 
   pub platform_constraint: Option<Platform>,
 
-  pub is_nailgunnable: bool,
+  ///
+  /// If non-empty, the Digest of a nailgun server to use to attempt to spawn the Process.
+  ///
+  /// TODO: Currently this Digest must be a subset of the `input_digest`, but we should consider
+  /// making it disjoint, and then automatically merging it.
+  ///
+  pub use_nailgun: Digest,
```

Comment:
On the Python side this is a default argument, so users mostly don't need to set it. But we have in general used the "null object" (with EMTY_DIGEST as null) pattern for digests, rather than making them optional on Process.